File: app/src/main/java/com/yoneya1014/jetfarm_test1/ShowSavedDataActivity.java
package com.yoneya1014.jetfarm_test1;

import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.content.Context;
import android.content.DialogInterface;
import android.content.Intent;
import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.util.Base64;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.DocumentReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;

import java.text.SimpleDateFormat;
import java.util.Locale;

public class ShowSavedDataActivity extends AppCompatActivity {

    private DocumentReference docRef;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.showsaveddata);
        Intent intent = getIntent();
        String dataName = intent.getStringExtra("dataname");
        setTitle(dataName);
        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        final ProgressDialog progressDialog = new ProgressDialog(this);
        progressDialog.setMessage("お待ちください");
        progressDialog.setCancelable(false);
        progressDialog.setIndeterminate(false);
        progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
        progressDialog.show();
        if (netWorkCheck(this)) {
            final TextView temp = findViewById(R.id.textView5);
            final TextView humid = findViewById(R.id.textView6);
            final TextView camera = findViewById(R.id.textView7);
            final TextView time = findViewById(R.id.savedtime);
            final ImageView imageView = findViewById(R.id.imageView2);
            docRef = FirebaseFirestore.getInstance().collection("userSavedData").document(dataName);
            docRef.get().addOnCompleteListener(new OnCompleteListener<DocumentSnapshot>() {
                @Override
                public void onComplete(@NonNull Task<DocumentSnapshot> task) {
                    if (task.isSuccessful()) {
                        DocumentSnapshot document = task.getResult();
                        if (document.exists()) {
                            SimpleDateFormat sdf = new SimpleDateFormat("yyyy年MM月dd日 HH時mm分ss秒", Locale.JAPAN);
                            temp.setText(String.format("温度：%s℃", document.getDouble("temp").toString()));
                            humid.setText(String.format("湿度：%s％", document.getDouble("humid").toString()));
                            time.setText(String.format("保存日時：%s", sdf.format(document.getTimestamp("timestamp").toDate())));
                            camera.setText(String.format(Locale.JAPAN, "カメラ番号：%1.0f", document.getDouble("cameranumber")));
                            byte[] imagearray = Base64.decode(document.getString("base64image"), Base64.DEFAULT);
                            Bitmap bitmap = BitmapFactory.decodeByteArray(imagearray, 0, imagearray.length);
                            imageView.setImageBitmap(bitmap);
                        }
                    }
                    progressDialog.dismiss();
                }
            });
        } else {
            AlertDialog.Builder builder = new AlertDialog.Builder(getApplicationContext());
            builder.setTitle("エラー")
                    .setCancelable(false)
                    .setMessage("コネクションの確立に失敗しました")
                    .setPositiveButton("OK", new DialogInterface.OnClickListener() {
                        @Override
                        public void onClick(DialogInterface dialog, int which) {
                            progressDialog.dismiss();
                            finish();
                        }
                    });
        }
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.showsaveddatamenu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
            case R.id.menu6:
                final ProgressDialog progressDialog = new ProgressDialog(this);
                progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
                progressDialog.setIndeterminate(false);
                progressDialog.setMessage("データ削除中");
                progressDialog.setCancelable(false);
                progressDialog.show();
                docRef.delete()
                        .addOnCompleteListener(new OnCompleteListener<Void>() {
                            @Override
                            public void onComplete(@NonNull Task<Void> task) {
                                if (task.isSuccessful()) {
                                    SavedDataSelectMenuActivity.deleteFrag = true;
                                    progressDialog.dismiss();
                                    finish();
                                    Toast.makeText(getApplicationContext(), "データ削除完了", Toast.LENGTH_LONG).show();
                                }
                            }
                        });
                break;
        }
        return super.onOptionsItemSelected(item);
    }

    private boolean netWorkCheck(Context context) {
        ConnectivityManager cm = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
        NetworkInfo info = cm.getActiveNetworkInfo();
        if (info != null) {
            return info.isConnected();
        } else {
            return false;
        }
    }
}

File: app/src/main/java/com/yoneya1014/jetfarm_test1/ShowSavedDataActivity.kt
package com.yoneya1014.jetfarm_test1

import android.app.AlertDialog
import android.app.ProgressDialog
import android.content.Context
import android.graphics.BitmapFactory
import android.net.ConnectivityManager
import android.os.Bundle
import android.util.Base64
import android.view.Menu
import android.view.MenuItem
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.google.firebase.firestore.DocumentReference
import com.google.firebase.firestore.FirebaseFirestore
import java.text.SimpleDateFormat
import java.util.*

class ShowSavedDataActivity : AppCompatActivity() {

    private var docRef: DocumentReference? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.showsaveddata)
        val intent = intent
        val dataName = intent.getStringExtra("dataname")
        title = dataName
        val actionBar = supportActionBar
        actionBar!!.setDisplayHomeAsUpEnabled(true)
        val progressDialog = ProgressDialog(this)
        progressDialog.setMessage("お待ちください")
        progressDialog.setCancelable(false)
        progressDialog.isIndeterminate = false
        progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER)
        progressDialog.show()
        if (netWorkCheck(this)) {
            val temp = findViewById<TextView>(R.id.textView5)
            val humid = findViewById<TextView>(R.id.textView6)
            val camera = findViewById<TextView>(R.id.textView7)
            val time = findViewById<TextView>(R.id.savedtime)
            val imageView = findViewById<ImageView>(R.id.imageView2)
            docRef = FirebaseFirestore.getInstance().collection("userSavedData").document(dataName)
            docRef!!.get().addOnCompleteListener { task ->
                if (task.isSuccessful) {
                    val document = task.result
                    if (document!!.exists()) {
                        val sdf = SimpleDateFormat("yyyy年MM月dd日 HH時mm分ss秒", Locale.JAPAN)
                        temp.text = String.format("温度：%s℃", document.getDouble("temp")!!.toString())
                        humid.text = String.format("湿度：%s％", document.getDouble("humid")!!.toString())
                        time.text = String.format("保存日時：%s", sdf.format(document.getTimestamp("timestamp")!!.toDate()))
                        camera.text = String.format(Locale.JAPAN, "カメラ番号：%1.0f", document.getDouble("cameranumber"))
                        val imagearray = Base64.decode(document.getString("base64image"), Base64.DEFAULT)
                        val bitmap = BitmapFactory.decodeByteArray(imagearray, 0, imagearray.size)
                        imageView.setImageBitmap(bitmap)
                    }
                }
                progressDialog.dismiss()
            }
        } else {
            val builder = AlertDialog.Builder(applicationContext)
            builder.setTitle("エラー")
                    .setCancelable(false)
                    .setMessage("コネクションの確立に失敗しました")
                    .setPositiveButton("OK") { _, _ ->
                        progressDialog.dismiss()
                        finish()
                    }
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        val inflater = menuInflater
        inflater.inflate(R.menu.showsaveddatamenu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
            R.id.menu6 -> {
                val progressDialog = ProgressDialog(this)
                progressDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER)
                progressDialog.isIndeterminate = false
                progressDialog.setMessage("データ削除中")
                progressDialog.setCancelable(false)
                progressDialog.show()
                docRef!!.delete()
                        .addOnCompleteListener { task ->
                            if (task.isSuccessful) {
                                SavedDataSelectMenuActivity.deleteFrag = true
                                progressDialog.dismiss()
                                finish()
                                Toast.makeText(applicationContext, "データ削除完了", Toast.LENGTH_LONG).show()
                            }
                        }
            }
        }
        return super.onOptionsItemSelected(item)
    }

    private fun netWorkCheck(context: Context): Boolean {
        val cm = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val info = cm.activeNetworkInfo
        return info?.isConnected ?: false
    }
}
